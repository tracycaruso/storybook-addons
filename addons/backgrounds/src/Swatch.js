import React from 'react';
import PropTypes from 'prop-types';

import styled from '@emotion/styled';

const Button = styled.button(({ theme }) => ({
  listStyle: 'none',
  backgroundColor: theme.barFill,
  textAlign: 'center',
  border: theme.mainBorder,
  borderRadius: theme.mainBorderRadius,
  color: 'inherit',
  cursor: 'pointer',
  display: 'inline-block',
  width: 175,
  verticalAlign: 'top',
  wordWrap: 'break-word',
  padding: 0,
  overflow: 'hidden',
}));

const Block = styled.div(({ bg, theme }) => ({
  height: 80,
  transition: 'opacity 0.25s ease-in-out',
  borderBottom: theme.mainBorder,
  background: bg,
  backgroundSize: 'cover',
  backgroundPosition: 'center',
}));

const Box = styled.div({
  listStyle: 'none',
  paddingLeft: 10,
  paddingRight: 10,
});

const Name = styled.h4({
  float: 'left',
  fontWeight: 'bold',
});
const Value = styled.h4({
  float: 'right',
  fontWeight: 'normal',
});

const Swatch = ({ name, value, setBackground }) => (
  <Button onClick={() => setBackground(value)} onMouseDown={event => event.preventDefault()}>
    <Block bg={value} />
    <Box>
      <Name>{name}:</Name>
      <Value>
        <em>{value}</em>
      </Value>
    </Box>
  </Button>
);
Swatch.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  setBackground: PropTypes.func.isRequired,
};

export default Swatch;
