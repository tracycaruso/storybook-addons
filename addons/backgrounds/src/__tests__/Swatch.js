import React from 'react';
import { shallow, mount } from 'enzyme';

import Swatch from '../Swatch';

const mockedSetBackround = jest.fn();

describe('Swatch', () => {
  it('should exist', () => {
    const swatch = shallow(<Swatch value="bar" name="foo" setBackground={mockedSetBackround} />);

    expect(swatch).toBeDefined();
  });

  it('should render the name of the swatch', () => {
    const markup = shallow(
      <Swatch value="bar" name="foo" setBackground={mockedSetBackround} />
    ).html();

    expect(markup.match(/foo/gim)).toHaveLength(1);
  });

  it('should render the value of the swatch and set it to be the background', () => {
    const result = shallow(<Swatch value="bar" name="foo" setBackground={mockedSetBackround} />);

    expect(result).toMatchSnapshot();
  });

  it('should emit message on click', () => {
    const spy = jest.fn();
    const swatch = mount(<Swatch value="#e6e6e6" name="Gray" setBackground={spy} />);
    swatch.simulate('click');

    expect(spy).toBeCalledWith('#e6e6e6');
  });
});
