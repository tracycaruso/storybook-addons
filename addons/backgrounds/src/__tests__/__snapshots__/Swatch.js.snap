// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swatch should render the value of the swatch and set it to be the background 1`] = `
<ForwardRef
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <ForwardRef
    bg="bar"
  />
  <ForwardRef>
    <ForwardRef>
      foo
      :
    </ForwardRef>
    <ForwardRef>
      <em>
        bar
      </em>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;
